src/config/states.ts

```diff
@@ -0,0 +1,340 @@
+import { AppointmentDetails, CallData } from './types';
+
+// ----------------------------------------------------------------------------- //
+// State Machine Implementation for Appointment Management                       //
+// ----------------------------------------------------------------------------- //
+// This file defines a complete state machine for managing outbound appointment  //
+// confirmations and rescheduling calls. The structure is designed to be:        //
+//    1. Type-safe: Using TypeScript enums and interfaces                        //
+//    2. Declarative: States and transitions are clearly defined                 //
+//    3. Maintainable: All state definitions in one place                        //
+//    4. Extensible: Easy to add new states or actions                           //
+//                                                                               //
+// HOW TO USE THIS PATTERN IN YOUR OWN APP:                                      //
+//    1. Define your states using StateEnum                                      //
+//    2. Define your actions using ActionEnum                                    //
+//    3. Create state definitions including templates and actions                //
+//    4. Use the transition function in stateMachine.ts to move between states   //
+// ----------------------------------------------------------------------------- //
+
+/**
+ * StateEnum - All possible states in the appointment management flow
+ * 
+ * Use this enum to reference states throughout your application.
+ * Add new states here first, then create its definition below.
+ */
+export enum StateEnum {
+  INITIAL = 'initial',
+  IDENTITY_CHECKING = 'identity_checking',
+  ATTENDANCE_CHECKING = 'attendance_checking',
+  RESCHEDULE_OFFERING = 'reschedule_offering',
+  RESCHEDULE_CONFIRMED = 'reschedule_confirmed',
+  APPOINTMENT_CANCELLED = 'appointment_cancelled',
+  CALL_ENDED = 'call_ended'
+}
+
+/**
+ * ActionEnum - All possible actions that can trigger state transitions
+ * 
+ * These are the "events" that trigger transitions between states.
+ * Each state defines which subset of actions it supports.
+ */
+export enum ActionEnum {
```

Comment:
Do we need an action enum at all? If the names are local to each state then maybe not?

Reply:
we could remove it but i think it's nice to see all the actions defined in one place.

Reply:
i will keep it in there for now and merge this. we can simplify later if we get feedback that it's too cumbersome.